rllib/connectors/common/add_states_from_episodes_to_batch.py

```diff
@@ -315,8 +316,18 @@ def __call__(
                         neg_index_as_lookback=True,
                     )
                 )
-                # state_outs.shape=(T,[state-dim])  T=episode len
-                state_outs = sa_episode.get_extra_model_outputs(key=Columns.STATE_OUT)
+                if Columns.STATE_OUT in sa_episode.extra_model_outputs:
+                    # state_outs.shape=(T,[state-dim])  T=episode len
+                    state_outs = sa_episode.get_extra_model_outputs(
+                        key=Columns.STATE_OUT
+                    )
+                else:
+                    state_outs = tree.map_structure(
```

Comment:
Can we explain here why we would repeat the lookback state `len(episode)` times? What's the logic behind doing this?

Reply:
Good point. The main case where this happens is in offline learning when the expert was non-stateful. I have added some comments to make it more clear.